### DISCUSSION TITLE
CLI errors when creating new project

### DISCUSSION THREAD

**Discussion by mandofever78** [quasarframework/quasar#14373] [CLI - PWA mode] [answered]
I am attempting to create a new Quasar (webpack) project using the latest version of the CLI and the process will not complete without errors. I have uninstalled and reinstalled the CLI and have attempted to spin up v1 and v2 projects with various options to no avail. Every attempt results in the following errors being thrown:

`
Quasar • Generating files...

 - README.md
 - .editorconfig
 - .gitignore
 - .postcssrc.js
 - package.json
(node:15184) UnhandledPromiseRejectionWarning: SyntaxError: Unexpected token n in JSON at position 154
    at JSON.parse (<anonymous>)
    at Object.module.exports.renderTemplate (/Users/nolanlawrence/.config/yarn/global/node_modules/create-quasar/utils/index.js:82:31)
    at module.exports (/Users/nolanlawrence/.config/yarn/global/node_modules/create-quasar/templates/app/quasar-v2/ts-webpack/index.js:65:9)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async module.exports (/Users/nolanlawrence/.config/yarn/global/node_modules/create-quasar/templates/app/quasar-v2/index.js:29:3)
    at async module.exports (/Users/nolanlawrence/.config/yarn/global/node_modules/create-quasar/templates/app/index.js:9:3)
    at async run (/Users/nolanlawrence/.config/yarn/global/node_modules/create-quasar/index.js:70:3)
(node:15184) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:15184) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
✨  Done in 33.26s.
`

I've searched the forums and google extensively and can't find anything to point me in the right direction. Currently using:
node v 12.22.1
Quasar CLI v 1.3.2
yarn v 1.22.17

**Comment by pdanpdan** [comment]
What command are you using to create the project? What options did you select? What os and what shell are you using?

**Reply by mandofever78** [reply]
Creation command: 'yarn create quasar' 
I have attempted creation using all available "App with Quasar CLI" options in at least a dozen different configurations (Webpack only, as Vite requires node 14) including:
Quasar v1 & v2
Typescript & Javascript, 
Composition API ( with and without script setup) & Options API
Sass w SCSS & Indentation
With & without including Eslint
With & without including Axios

OS: MacOS Monterrey v 12.3.1
bash shell v 5.1.12(1)-release

**Comment by pdanpdan** [comment]
It looks like now it requires node v14
@rstoenescu (ref https://quasar.dev/start/quasar-cli#tl-dr)

```bash
~/Work/Tests
nvm use 12
Now using Node v12.22.12 (npm 8.11.0)

~/Work/Tests
yarn create quasar                                                                                                                                                     
yarn create v1.22.18
[1/4] Resolving packages...
[2/4] Fetching packages...
error vscode-jsonrpc@8.0.0-next.7: The engine "node" is incompatible with this module. Expected version ">=14.0.0". Got "12.22.12"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/create for documentation about this command.
```

**Reply by mandofever78** [reply]
I was about to mark your reply as the answer as I used nvm last week to temporarily change my node version to 14.20 and was able to use the cli to generate a new project. A week later, I have now set the default node version on my system to 14.20 and I am once again having the same error issues when attempting to generate a new project using the same methods and options as described above.

**Reply by mandofever78** [reply]
Here is the output when using
OS: MacOS Monterrey v 12.3.1
bash shell v 5.1.12(1)-release
Node 14.20
Quasar CLI v 1.3.2
Yarn v 1.22.17

`
$ yarn create quasar
yarn create v1.22.17
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
success Installed "create-quasar@1.1.0" with binaries:
      - create-quasar


 .d88888b.
d88P" "Y88b
888     888
888     888 888  888  8888b.  .d8888b   8888b.  888d888
888     888 888  888     "88b 88K          "88b 888P"
888 Y8b 888 888  888 .d888888 "Y8888b. .d888888 888
Y88b.Y8b88P Y88b 888 888  888      X88 888  888 888
 "Y888888"   "Y88888 "Y888888  88888P' "Y888888 888
       Y8b

✔ What would you like to build? › App with Quasar CLI, let's go!
✔ Project folder: … quasar-project
✔ Pick Quasar version: › Quasar v2 (Vue 3 | latest and greatest)
✔ Pick script type: › Typescript
✔ Pick Quasar App CLI variant: › Quasar App CLI with Vite
✔ Package name: … quasar-project
✔ Project product name: (must start with letter if building mobile apps) … Quasar App
✔ Project description: … A Quasar Project
✔ Author: … \"Tester McTesterson\" <"tmtillucm@icloud.com">
✔ Pick a Vue component style: › Composition API with <script setup>
✔ Pick your CSS preprocessor: › Sass with indented syntax
✔ Check the features needed for your project: › ESLint
✔ Pick an ESLint preset: › Prettier

 Quasar • Generating files...

 - README.md
 - .editorconfig
 - .gitignore
 - .npmrc
 - package.json
(node:19068) UnhandledPromiseRejectionWarning: SyntaxError: Unexpected token n in JSON at position 154
    at JSON.parse (<anonymous>)
    at Object.module.exports.renderTemplate (/Users/nolanlawrence/.config/yarn/global/node_modules/create-quasar/utils/index.js:82:31)
    at module.exports (/Users/nolanlawrence/.config/yarn/global/node_modules/create-quasar/templates/app/quasar-v2/ts-vite/index.js:65:9)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async module.exports (/Users/nolanlawrence/.config/yarn/global/node_modules/create-quasar/templates/app/quasar-v2/index.js:29:3)
    at async module.exports (/Users/nolanlawrence/.config/yarn/global/node_modules/create-quasar/templates/app/index.js:9:3)
    at async run (/Users/nolanlawrence/.config/yarn/global/node_modules/create-quasar/index.js:70:3)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:19068) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:19068) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
✨  Done in 16.41s.
`

**Reply by pdanpdan** [reply]
Just for testing, can you please try
- `nvm install v16` and in node v16 `yarn create quasar`
- then `nvm use v14` and try to run `quasar dev`

**Comment by mandofever78** [comment, marked as answer]
After upgrading with `nvm install 16` (v16.80.0), the debug messaging improved upon running `yarn create quasar` to show an error like this:

```
- package.json
undefined:6
  "author": "\"Tester McTesterson\" <"tmtillucm@icloud.com">",
                                                                ^
``` 

It turns out when running `yarn create quasar` that  neglecting to enter a value for the author field results in the field being auto-populated with the user values from the .gitconfig file. ie:

```
[user]
name = \"Tester McTesterson\"
email = \"tmtillucm@icloud.com\"

```

Putting a string value ( ie: tmtillucm@icloud.com ) without quotes in the author prompt defeats the auto-population and allows the cli to properly complete the create process. Problem solved! 

Perhaps there should be some clarification of the prompt requirements or the error message or both. 

Thank you for the help pointing me in the right direction!
